app/models/tag.rb

```diff
@@ -451,11 +451,11 @@ def self.in_prompt_restriction(restriction)
 
   def self.by_name_without_articles(fieldname = "name")
     fieldname = "name" unless fieldname.match(/^([\w]+\.)?[\w]+$/)
-    order("case when lower(substring(#{fieldname} from 1 for 4)) = 'the ' then substring(#{fieldname} from 5)
+    order(Arel.sql("case when lower(substring(#{fieldname} from 1 for 4)) = 'the ' then substring(#{fieldname} from 5)
             when lower(substring(#{fieldname} from 1 for 2)) = 'a ' then substring(#{fieldname} from 3)
             when lower(substring(#{fieldname} from 1 for 3)) = 'an ' then substring(#{fieldname} from 4)
             else #{fieldname}
-            end")
+            end"))
```

Comment:
Does `TagSetAssociation.by_name_without_articles` need similar treatment?
https://github.com/otwcode/otwarchive/blob/6dc754890a5cb9fb7da38a46fcf917998db3e63c/app/models/tagset_models/tag_set_association.rb#L15-L23
(Is raw SQL in `order` deprecated in 5.1 and needed to upgrade to 5.2, or is it deprecated in 5.2 and needed to upgrade to 6.0?)

Reply:
Possibly! I fixed them as I noticed them in the test suite output, so either I haven't noticed the one for `TagSetAssociation` or (frightening prospect) there is no test that exercises that method. (I prefer thinking that I haven't noticed that deprecation warning. 😓 )

Using "non-attribute arguments" in query methods[1] is a deprecation warning in Rails 5.2, and will become an exception in Rails 6.0. So it's not strictly necessary to upgrade to Rails 5.2, but I'm trying to put us ahead of the game (because that's what Elz was doing in her original branch 😅).

[1] The official verbiage is "Non-attribute arguments will be disallowed in Rails 6.0. This method should not be called with user-provided values, such as request parameters or model attributes. Known-safe values can be passed by wrapping them in Arel.sql()."

I still haven't fully digested that paragraph. If it's not best practice to use model attributes (because they can contain user input), why are they saying "non-attribute" arguments will be disallowed? Aren't model attributes "attribute arguments"? I find it very confusing.

Also, wrapping them in `Arel.sql()` isn't making anyone safer, since they say "known-safe values" - so it's still on us to be sure that those are safe values. (I think they are in this case, because we'll have done the sanitisation before persisting them.)

Reply:
It's not a good idea to use the values of model attributes unquoted, but the names of the attributes are generally fine. That's most likely what they mean by non-attribute arguments -- arguments that don't look like standard attribute names (aka column names).